driver/src/main/java/org/neo4j/driver/internal/InternalResultCursor.java

```diff
@@ -77,6 +222,9 @@ public int index( String key )
 
     public List<String> keys()
     {
+        while (keys == null && !done) {
```

Comment:
Only if the server responds incorrectly and in that case and Exception will be thrown, breaking out of the loop.